File: src/losses.py
import numpy as np
import cvxpy as cp


def l2_loss(x, y, beta):
    return cp.norm2(x @ beta - y)**2 / 2

def poisson_loss(x, y, beta):
    n = x.shape[0]
    return cp.sum(cp.exp(x @ beta) - cp.multiply(y , x @ beta))   ##/n make l1 l2 large

def logit_loss(x, y, beta):
    n = x.shape[0]
    loglik = cp.sum(cp.multiply(y, x @ beta) - cp.logistic(x @ beta))
    return -loglik/n

File: losses.R
library(CVXR)

l2_loss = function(x, y, beta){
  return (norm2(x %*% beta - y)**2 /2)
}

poisson_loss = function(x, y, beta){
  n = dim(x)[1]
  return (sum(exp(x %*% beta) - multiply(y,x %*% beta)))
}

logit_loss = function(x, y, beta){
  n = dim(x)[1]
  loglik = sum(multiply(y,x %*% beta) - logistic(x %*% beta))
  return (-loglik / n)
}